framework/src/timesteppers/SolutionTimeAdaptiveDT.C

```diff
@@ -52,15 +55,14 @@ SolutionTimeAdaptiveDT::~SolutionTimeAdaptiveDT()
 void
 SolutionTimeAdaptiveDT::step()
 {
-  gettimeofday(&_solve_start, NULL);
+  auto solve_start = std::chrono::system_clock::now();
 
   TimeStepper::step();
 
   if (converged())
   {
-    gettimeofday (&_solve_end, NULL);
-    double elapsed_time = (static_cast<double>(_solve_end.tv_sec  - _solve_start.tv_sec) +
-                           static_cast<double>(_solve_end.tv_usec - _solve_start.tv_usec)*1.e-6);
+    auto solve_end = std::chrono::system_clock::now();
+    auto elapsed_time = std::chrono::duration_cast<std::chrono::milliseconds>(solve_end - solve_start).count();
```

Comment:
Any idea what would happen if a timestep were to take _weeks_ to complete?  How long would the timestep have to be before this would overflow?

I don't think it's a big deal (probably not worth worrying about).